docs/input_processing.md

```diff
@@ -1,25 +1,17 @@
-As of now two inputs are supported:
-- JSON Schema Draft 7
-- AsyncAPI version 2.0.0
+# Input process
 
-## Internal model representation
-
-![Class diagram](./assets/class_diagram.png)
-
-As seen on the class diagram the `InputProcessor` is our main point of entry for processing input data. 
-
-It uses the defined input processors (`AsyncAPIInputProcessor`, `JsonSchemaInputProcessor`, ...) by first calling `shouldProcess` function of each and if the function returns true it calls the `process` function. 
+The [InputProcessor](../src/processors/InputProcessor.ts) is our main point of entry for processing input data. It uses the defined input processors (`AsyncAPIInputProcessor`, `JsonSchemaInputProcessor`, ...) by first calling `shouldProcess` function for each input processor to figure out if the input data should be processed. Once a processor have been found it 
 
 If no processes returns true it defaults to `JsonSchemaInputProcessor`. 
 
 The `process` function are expected to return `CommonInputModel` which is a wrapper for the core data representation of `CommonModel`. 
 
 This is done to ensure we can return multiple models for any input to allow for references, inheritance etc. 
 
-As said the core internal representation of a data model is `CommonModel`. This contains the data definition by using known keywords from JSON Schema, but instead of it representing a validation rules it represent data definition. The explanation for the `CommonModel` properties can be found [here](../API.md#CommonModel).
+As said the core internal representation of a data model is `CommonModel`. This contains the data definition by using known keywords from JSON Schema, but instead of it representing a validation rules it represent data definition. The explanation for the `CommonModel` properties can be found [here](../API.md##CommonModel).
```

Comment:
```suggestion
As said the core internal representation of a data model is `CommonModel`. It contains the data definition by using known keywords from JSON Schema, but instead of it representing a validation rules it represents data definition. The explanation for the `CommonModel` properties can be found [here](../API.md##CommonModel).
```

I'm not sure with `represents`